File: WpCoreSolution/Presentation/Wp.Web.Api/Areas/Admin/Models/Security/ClaimRoleModel.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Wp.Service.Security;

namespace Wp.Web.Api.Areas.Admin.Models
{
    public class ClaimRoleModel
    {
        public ClaimRoleModel()
        {
            AvailableClaims = new List<ClaimRecord>();
            AvailableRoles = new List<RoleModel>();
            Allowed = new Dictionary<string, IDictionary<string, bool>>(StringComparer.OrdinalIgnoreCase);
        }

        public IList<ClaimRecord> AvailableClaims { get; set; }
        public IList<RoleModel> AvailableRoles { get; set; }
        public IDictionary<string, IDictionary<string, bool>> Allowed { get; set; }
    }


}

File: wp-be/src/app/configuration/access-control-list/claim-role.model.ts
import { RoleModel } from '../../security/user/user.model';

export class ClaimRoleModel {
    availableClaims: ClaimRecord[];
    availableRoles: RoleModel[];
    allowed: { [key: string]: {[key: string]: boolean} }
    //allowed: [][];
}

export class ClaimRecord {
    claimType: string;
    claimValue: string;
}

